### PR TITLE
Validation webhook should ensure in UriRegexRewrite both match and rewrite fields present

### PR DESCRIPTION
**Pull request by wujiaqi**
**Please provide a description of this PR:**
Per PR comments in https://github.com/istio/istio/pull/45398, the Validation webhook should check for presence of all required fields in UriRegexRewrite

### CODE REVIEW

**Review by howardjohn — COMMENTED** [review]
Thanks!!

sorry forgot about this the first comment: can we also do validateStringMatchRegexp? Not the exact same since we are just a `string`, but the same check of size is less than 1024 and `regexp.Compile` works

**Comment by wujiaqi** [comment]
> sorry forgot about this the first comment: can we also do validateStringMatchRegexp? Not the exact same since we are just a `string`, but the same check of size is less than 1024 and `regexp.Compile` works

@howardjohn done. That one got a little weird, please see my comments. Here's my validations though locally

in my test VS yaml file, the backslashes are escaped (without escaped backslashes it's rejected during YAML --> JSON conversion)
```YAML
 33     rewrite:
 34       uriRegexRewrite:
 35         match: "^/service/([^/]+)(/.*)$"
 36         rewrite: "\\2/instance/\\1"
```

After being written to the cluster, and a `describe vs`
```
    Rewrite:
      Uri Regex Rewrite:
        Match:    ^/service/([^/]+)(/.*)$
        Rewrite:  \2/instance/\1
```

And in Envoy, I see this in the JSON dump
```JSON
"regex_rewrite": {
  "pattern": {
    "regex": "^/service/([^/]+)(/.*)$"
  },
  "substitution": "\\2/instance/\\1"
}
```

File: pkg/config/validation/validation_test.go [hunk 1]
```diff
@@ -2127,6 +2127,59 @@ func TestValidateHTTPRewrite(t *testing.T) {
...
+		{
+			name:  "uriRegexRewrite nil",
+			in:    nil,
+			valid: true,
+		},
+		{
+			name: "uriRegexRewrite happy path",
+			in: &networking.RegexRewrite{
+				Match:   "^/service/([^/]+)(/.*)$",
+				Rewrite: "\\2/instance/\\1",
```
Comment (howardjohn) [inline pkg/config/validation/validation_test.go:2145]:
```suggestion
				Rewrite: `\2/instance/\1`,
```

If you don't want to worry about escaping

File: pkg/config/validation/validation.go [hunk 2]
```diff
@@ -3180,9 +3183,28 @@ func validateHTTPRewrite(rewrite *networking.HTTPRewrite) error {
...
+	}
+	if regexRewrite.Match == "" || regexRewrite.Rewrite == "" {
+		return errors.New("UriRegexRewrite requires both Rewrite and Match fields to be specified")
+	}
+	var err error
+	// In the validation context we cannot directly run the Match and Rewrite strings through regexp.Compile() because
+	// they are still strings at this point. We need to verify that they would be valid RE2 regex after they are written
+	// to Envoy route, hence escape the backslashes
+	err = errors.Join(err, validateStringRegexp(strings.ReplaceAll(regexRewrite.Match, "\\", "\\\\"), "HTTPRewrite.UriRegexRewrite.Match"))
+	err = errors.Join(err, validateStringRegexp(strings.ReplaceAll(regexRewrite.Rewrite, "\\", "\\\\"), "HTTPRewrite.UriRegexRewrite.Rewrite"))
```
Comment (howardjohn) [inline pkg/config/validation/validation.go:3204]: I am not sure this is true. 

YAML input: 
```yaml
rewrite: "\\2"
```

In go is the string

```
rewrite := `\2`
```

Which is the yaml we want to validate against. We should pass as-is

Comment (wujiaqi) [reply pkg/config/validation/validation.go:3204]: my bad, I see what the issue was. I was treating the Rewrite string as a regex pattern when it's actually just a string with substitution placeholders  e.g. `\2`

**Review by howardjohn — APPROVED** [review]

File: pkg/config/validation/validation.go (hunk 2 above)
Comment (howardjohn) [reply pkg/config/validation/validation.go]: ahhh... right. Thanks!!

**Pull request merged**